src/hotspot/cpu/s390/c1_Runtime1_s390.cpp

```diff
@@ -589,6 +589,76 @@ OopMapSet* Runtime1::generate_code_for(C1StubId id, StubAssembler* sasm) {
       __ z_br(Z_R14);
     }
     break;
+    case C1StubId::is_instance_of_id:
+    {
+      // Mirror: Z_ARG1(R2)
+      // Object: Z_ARG2
+      // Temps: Z_ARG3, Z_ARG4, Z_ARG5, Z_tmp_1, Z_tmp_2
+      // Result: Z_RET(R2)
+
+      // Get the Klass* into Z_ARG3
+      Register klass = Z_ARG3 , obj = Z_ARG2, result = Z_RET;
+      Register temp0 = Z_ARG4, temp1 = Z_ARG5, temp2 = Z_tmp_1, temp3 = Z_tmp_2;
```

Comment:
```suggestion
      Register temp0 = Z_ARG4, temp1 = Z_ARG5, temp2 = Z_R11, temp3 = Z_R12;
```
Aliasing `temp2` to `Z_tmp_1` is just too confusing to the reader.

Reply:
Updated. 

`Z_tmp_1` and `Z_tmp_2` refers to `Z_R10` and `Z_R11` respectively, So I have update the patch according to that. 

```cpp
constexpr Register Z_tmp_1 =  Z_R10;
constexpr Register Z_tmp_2 =  Z_R11;
```